### PR TITLE
New parameter for `sync-agent-groups-get` command to filter recently registered agents

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#12332|

## Description

This PR adds a non mandatory parameter to the `sync-agent-groups-get` command to select only the agents which are registered since some time ago.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- [x] Scan build

- [x] Added unit tests (for new features)

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Scan build

No bugs found

scan_build

### Manual tests

These are the agents registered for the test, with their corresponding **date_add** values and groups

current_state

**Condition all**

condition_all

**Condition all and delta**

condition_all_delta

**Condition sync_status**

condition_sync_status

**Condition sync_status and delta**

condition_sync_status_delta

### Coverage 

**wdb_parse_global_sync_agent_groups_get**

global_sync_agent_groups_get

**wdb_global_sync_agent_groups_get**

wdb_global_sync_agent_groups_get

File: src/wazuh_db/wdb_global.c [hunk 1]
```diff
@@ -1402,6 +1402,8 @@ wdbc_result wdb_global_sync_agent_groups_get(wdb_t *wdb, wdb_groups_sync_conditi
     char *out_aux = cJSON_PrintUnformatted(*output);
     size_t response_size = strlen(out_aux);
     os_free(out_aux);
+    // Recent agents will be excluded
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_global.c:1405]: I would reword this comment to `Agents registered recently may be excluded depending on the 'agent_registration_delta' value.`

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Just a few comments.

File: src/wazuh_db/wdb_global.c [hunk 2]
```diff
@@ -1402,6 +1402,8 @@ wdbc_result wdb_global_sync_agent_groups_get(wdb_t *wdb, wdb_groups_sync_conditi
...
+    int agent_registration_time = time(NULL) - agent_registration_delta;
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_db/wdb_global.c:1406]: You are right!
Fixed

File: src/wazuh_db/wdb_global.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_global.c:1405]: Fixed!

**Review by DProvinciani — APPROVED** [review]
Looks good!
